lib/auth/github.go

```diff
@@ -371,6 +371,7 @@ func validateGithubAuthCallbackHelper(ctx context.Context, m githubManager, diag
 
 	auth, err := m.validateGithubAuthCallback(ctx, diagCtx, q)
 	diagCtx.Info.Error = trace.UserMessage(err)
+	event.AppliedLoginRules = diagCtx.Info.AppliedLoginRules
```

Comment:
nit: Move this to the struct initialization above?

```diff
	event := &apievents.UserLogin{
		Metadata: apievents.Metadata{
			Type: events.UserLoginEvent,
		},
		Method:            events.LoginMethodGithub,
+		AppliedLoginRules: diagCtx.Info.AppliedLoginRules,
	}
```

Reply:
so this is kind of ugly but `diagCtx.Info.AppliedLoginRules` doesn't get set until the `m.validateGithubAuthCallback` call

Reply:
Ohh, could you sneak in a comment into some other PR saying that, if you can? That's quite surprising.

Reply:
I agree it smells, but basically this entire function copies from `diagCtx` to set fields in the event, I don't think this instance really needs a special comment for it